rcl/src/rcl/arguments.c

```diff
@@ -883,21 +883,30 @@ rcl_remove_ros_arguments(
   int * nonros_argc,
   const char ** nonros_argv[])
 {
-  RCL_CHECK_ARGUMENT_FOR_NULL(argv, RCL_RET_INVALID_ARGUMENT);
-  RCL_CHECK_ARGUMENT_FOR_NULL(nonros_argc, RCL_RET_INVALID_ARGUMENT);
   RCL_CHECK_ARGUMENT_FOR_NULL(args, RCL_RET_INVALID_ARGUMENT);
   RCL_CHECK_ALLOCATOR_WITH_MSG(&allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ARGUMENT_FOR_NULL(nonros_argc, RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ARGUMENT_FOR_NULL(nonros_argv, RCL_RET_INVALID_ARGUMENT);
+  if (NULL != *nonros_argv) {
+    RCL_SET_ERROR_MSG("Output nonros_argv pointer is not null. May leak memory.");
+    return RCL_RET_INVALID_ARGUMENT;
+  }
 
   *nonros_argc = rcl_arguments_get_count_unparsed(args);
-  *nonros_argv = NULL;
-
-  if (*nonros_argc <= 0) {
+  if (*nonros_argc < 0) {
```

Comment:
Consider including tests for these failure conditions as well.

Reply:
Done in https://github.com/ros2/rcl/pull/518/commits/002cd09784f157d24c1a58bc8d25c86b6516bbd4.